cyral/resource_cyral_sidecar.go

```diff
@@ -184,11 +240,90 @@ func getSidecarDataFromResource(c *client.Client, d *schema.ResourceData) (Sidec
 	for i, label := range labels {
 		sidecarDataLabels[i] = (label).(string)
 	}
-	return SidecarData{
-		ID:              d.Id(),
-		Name:            d.Get("name").(string),
-		Labels:          sidecarDataLabels,
-		SidecarProperty: sp,
-		UserEndpoint:    d.Get("user_endpoint").(string),
+
+	cbs := getCertificateBundleSecret(d)
+
+	log.Printf("[DEBUG] end getSidecarDataFromResource")
+	return &SidecarData{
+		ID:                       d.Id(),
+		Name:                     d.Get("name").(string),
+		Labels:                   sidecarDataLabels,
+		SidecarProperty:          sp,
+		UserEndpoint:             d.Get("user_endpoint").(string),
+		CertificateBundleSecrets: *cbs,
 	}, nil
 }
+
+func flattenCertificateBundleSecrets(cbs *CertificateBundleSecrets) []interface{} {
+	log.Printf("[DEBUG] Init flattenCertificateBundleSecrets")
+	if cbs != nil {
+		if _, ok := (*cbs)["sidecar"]; ok {
+			flatCBS := make([]interface{}, 1)
+			cb := make(map[string]interface{})
+
+			contentCB := make([]interface{}, 0, len(*cbs))
+			for key, val := range *cbs {
+				// Ignore self-signed certificates
+				if key != "sidecar-generated-selfsigned" {
+					log.Printf("[DEBUG] key: %v", key)
+					log.Printf("[DEBUG] val: %v", val)
+
+					contentCBMap := make(map[string]interface{})
+					contentCBMap["secret_id"] = val.SecretId
+					contentCBMap["engine"] = val.Engine
+					contentCBMap["type"] = val.Type
+
+					contentCB = append(contentCB, contentCBMap)
+				}
+			}
+
+			cb["sidecar"] = contentCB
+			flatCBS[0] = cb
+
+			log.Printf("[DEBUG] end flattenCertificateBundleSecrets %v", flatCBS)
+			return flatCBS
+		}
+	}
+
+	log.Printf("[DEBUG] end flattenCertificateBundleSecrets")
+	return nil
+}
+
+func getCertificateBundleSecret(d *schema.ResourceData) *CertificateBundleSecrets {
+	log.Printf("[DEBUG] Init getCertificateBundleSecret")
+	rdCBS := d.Get("certificate_bundle_secrets").(*schema.Set).List()
+	ret := make(CertificateBundleSecrets)
+
+	if len(rdCBS) > 0 {
+		cbsMap := rdCBS[0].(map[string]interface{})
+		for k, v := range cbsMap {
+			vList := v.(*schema.Set).List()
+			// 1. k = "sidecar" or other direct internal elements of certificate_bundle_secrets
+			// 2. Also one element on this list due to MaxItems...
+			// 3. Ignore self signed certificates
+			if len(vList) > 0 && k != "sidecar-generated-selfsigned" {
```

Comment:
Notice that `k = "sidecar"` will always be true since the schema only supports certificate bundle secret for the `sidecar` key, so there's no need to check `k` value here.

Reply:
Not always. See comment in `1`. What I'm trying to do here is not mess with the `sidecar-generated-selfsigned` element as it does not matter for the provider, but all the others, including `sidecar` does. In this sense, I don't want the provider to break when we add a new type of `CertificateBundleSecret` with a different name. So far, only `sidecar-generated-selfsigned` and `sidecar` exist, but we need to be resilient to other types in the future.